src/Microsoft.AspNetCore.Razor.Language/Legacy/ParserContext.cs

```diff
@@ -29,7 +29,7 @@ public ParserContext(RazorSourceDocument source, RazorParserOptions options)
 
         public SyntaxTreeBuilder Builder { get; }
 
-        public ErrorSink ErrorSink { get; }
+        public ErrorSink ErrorSink { get; set; }
```

Comment:
This is super hackery. Essentially we need to swap out the underlying error sink when parsing directive tokens because the tokenizer/other parsing pieces can log their own errors which we still need to capture.